internal/generate/olm-catalog/csv_updaters.go

```diff
@@ -286,39 +278,19 @@ func (c manifestCollection) applyCustomResourceDefinitions(csv *operatorsv1alpha
 // updateDescriptions parses APIs in apisDir for code and annotations that
 // can build a verbose crdDescription and updates existing crdDescriptions in
 // csv. If no code/annotations are found, the crdDescription is appended as-is.
-func updateDescriptions(csv *operatorsv1alpha1.ClusterServiceVersion, apisDir string) error {
-	updatedDescriptions := []operatorsv1alpha1.CRDDescription{}
-	for _, currDescription := range csv.Spec.CustomResourceDefinitions.Owned {
-		group := currDescription.Name
-		if split := strings.Split(currDescription.Name, "."); len(split) > 1 {
-			group = strings.Join(split[1:], ".")
-		}
-		// Parse CRD descriptors from source code comments and annotations.
-		gvk := schema.GroupVersionKind{
-			Group:   group,
-			Version: currDescription.Version,
-			Kind:    currDescription.Kind,
-		}
-		newDescription, err := descriptor.GetCRDDescriptionForGVK(apisDir, gvk)
-		if err != nil {
-			if goerrors.Is(err, descriptor.ErrAPIDirNotExist) {
-				log.Infof("Directory for API %s does not exist. Skipping CSV annotation parsing for API.", gvk)
-			} else if goerrors.Is(err, descriptor.ErrAPITypeNotFound) {
-				log.Infof("No kind type found for API %s. Skipping CSV annotation parsing for API.", gvk)
-			} else {
-				// TODO: Should we ignore all CSV annotation parsing errors and simply log the error
-				// like we do for the above cases.
-				return fmt.Errorf("failed to set CRD descriptors for %s: %v", gvk, err)
-			}
-			// Keep the existing description and don't update on error
-			updatedDescriptions = append(updatedDescriptions, currDescription)
-		} else {
-			// Replace the existing description with the newly parsed one
-			newDescription.Name = currDescription.Name
-			updatedDescriptions = append(updatedDescriptions, newDescription)
-		}
+func updateDescriptions(csv *operatorsv1alpha1.ClusterServiceVersion,
+	apisDir string, opType projutil.OperatorType) error {
+
+	keys := make([]registry.DefinitionKey, len(csv.Spec.CustomResourceDefinitions.Owned))
+	for i, crd := range csv.Spec.CustomResourceDefinitions.Owned {
+		keys[i] = descriptor.CRDDescriptionToKey(crd)
+	}
+	// FEAT(estroz): customresourcedefinition should not be updated for
+	// Ansible and Helm CSV's until annotated updates are implemented.
+	switch opType {
+	case projutil.OperatorTypeGo:
```

Comment:
I think it has already the info in the docs that this feature is not supported for Helm and Ansible, however, we have the go-to make it work for them as well. So, why not just leave here without the type check? Then, it indeed makes easier we know what is missing or not be performed for the other types as well. WDYT?

Reply:
Markers don't work on non-Go files so this will fail if the project is non-Go.

Reply:
Ok. So, it makes completed sense.